src/main/java/org/codehaus/groovy/runtime/DefaultGroovyMethods.java

```diff
@@ -6644,6 +6286,18 @@ public static Object sum(Collection self, Object initialValue, Closure closure)
         }
         return result;
     }
+    /**
+     * Sums all the items in a stream. This is equivalent to invoking the
+     * "plus" method on all items in the stream.
+     * <pre class="groovyTestCase">assert 1+2+3+4 == [1,2,3,4].stream().sum()</pre>
+     *
+     * @param stream The stream containing the values to add together
+     * @param <T> The type of the values in the stream
+     * @return The sum of all of the items
+     */
+    public static <T> T sum(Stream<T> stream) {
```

Comment:
It's better to create JIRA ticket to track the improvement

Reply:
Sorry this was my first pull request to the groovy repository, can you explain to me how to do it?

Reply:
Here is the link to JIRA ;-)
https://issues.apache.org/jira/projects/GROOVY/summary